### DISCUSSION TITLE
"setup_expt.py" failed on Orion because of module named 'yaml'

### DISCUSSION THREAD

**Discussion by wzzheng90** [NOAA-EMC/global-workflow#1778] [General]
Hi,

I tried to run experiment generator script "./setup_expt.py" on Orion and got the following errors.
It looks like about the module named 'yaml'.  However, I do not find 'yaml' but 'yafyaml/0.5.1' is available.
After I loaded 'yafyaml/0.5.1', I got the same errors. 
What else should I set up on Orion before running "setup_expt.py"?  

==================
./setup_expt.py gfs forecast-only --pslot hr1TQ2m2022070400 --idate 2022070400 --edate 2022070400 --resdet 768 --gfs_cyc 1 --comrot /work/noaa/stmp/wzheng/HR2_test/COMROOT --expdir /work/noaa/stmp/wzheng/HR2_test/EXPDIR
Traceback (most recent call last):
  File "/work/noaa/fv3-cam/wzheng/git/HR2/global-workflow/workflow/./setup_expt.py", line 13, in <module>
    from hosts import Host
  File "/work/noaa/fv3-cam/wzheng/git/HR2/global-workflow/workflow/hosts.py", line 6, in <module>
    from wxflow import YAMLFile
  File "/work/noaa/fv3-cam/wzheng/git/HR2/global-workflow/workflow/wxflow/__init__.py", line 3, in <module>
    from .yaml_file import YAMLFile, parse_yaml, parse_yamltmpl, parse_j2yaml, save_as_yaml, dump_as_yaml, vanilla_yaml
  File "/work/noaa/fv3-cam/wzheng/git/HR2/global-workflow/workflow/wxflow/yaml_file.py", line 4, in <module>
    import yaml
ModuleNotFoundError: No module named 'yaml'

===============
Thanks,

Weizhong

**Comment by WalterKolczynski-NOAA** [comment]
Did you source `workflow/gw_setup.sh`?

**Reply by wzzheng90** [reply]
Thanks, Walter.   It works after I type this command!

**Reply by wzzheng90** [reply]
@WalterKolczynski-NOAA   Why I can not use this command (source workflow/gw_setup.sh) on Hera?   The error shows "Illegal variable name".  Thanks.

**Reply by WalterKolczynski-NOAA** [reply]
Works for me. Are you maybe not using bash on Hera?

**Reply by wzzheng90** [reply]
I see. I did use csh on Hera.    
Now I use bash and it works.  Thanks!!

**Comment by aerorahul** [comment]
user reports success after switching to bash.
